src/plugins/intel_npu/src/plugin/npuw/lazy_tensor.cpp

```diff
@@ -4,161 +4,219 @@
 
 #include "lazy_tensor.hpp"
 
-using ov::npuw::weights::ConcatMeta;
-using ov::npuw::weights::ConstPtr;
+#include <tuple>
+#include <type_traits>
+#include <variant>
+
+#include "logging.hpp"
+#include "openvino/runtime/make_tensor.hpp"
+#include "util.hpp"
+
 using ov::npuw::weights::LazyTensor;
-using ov::npuw::weights::OrigData;
-using ov::npuw::weights::Transform;
-using ov::npuw::weights::TransformType;
-using ov::npuw::weights::UnpackMeta;
 
 namespace ov {
 namespace npuw {
 namespace weights {
+namespace op {
+struct Tensor {
+    ov::Tensor tensor;
+};
+struct Const {
+    std::shared_ptr<ov::op::v0::Constant> node;
+};
+struct Concat {
+    std::vector<LazyTensor> tensors;
+    std::size_t axis;
+};
+struct Unpack {
+    LazyTensor w, z, s;
+    ov::element::Type type;
+    ov::Shape shape;
+};
+struct Permute {
+    LazyTensor tensor;
+    std::vector<std::size_t> axes;
+};
+struct Convert {
+    LazyTensor tensor;
+    ov::element::Type type;
+};
+}  // namespace op
+
+using Transform = std::variant<op::Tensor, op::Const, op::Concat, op::Unpack, op::Permute, op::Convert>;
 
 struct LazyTensorImpl {
 public:
     LazyTensorImpl() = default;
-    LazyTensorImpl(const TransformType& type, const Transform& transform);
-
-    bool operator==(const LazyTensorImpl& other) const;
+    explicit LazyTensorImpl(Transform&& t);
 
     ov::Tensor eval() const;
 
-    ov::Tensor get_orig_tensor() const;
-
+    bool operator==(const LazyTensorImpl& other) const;
     std::size_t get_hash() const;
+    const void* get_data() const;
```

Comment:
there's no data in LazyTensor until it is evaluated, what's the idea behind this one?

Reply:
Removed